### PR TITLE
Add ContractPermissionAttributes

### PR DESCRIPTION
**Pull request by shargon**
Close https://github.com/neo-project/neo-devpack-dotnet/issues/459

### CODE REVIEW

File: src/Neo.SmartContract.Framework/WildcardContractPermissionAttribute.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+using System;
+
+namespace Neo.SmartContract.Framework
+{
+    [AttributeUsage(AttributeTargets.Class, AllowMultiple = false)]
+    public class WildcardContractPermissionAttribute : Attribute
```
Comment (erikzhang) [inline src/Neo.SmartContract.Framework/WildcardContractPermissionAttribute.cs:6]: We don't need this. We can use `[ContractPermission("*", methods)]`.

File: src/Neo.SmartContract.Framework/ContractPermissionAttribute.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+using System;
+
+namespace Neo.SmartContract.Framework
+{
+    [AttributeUsage(AttributeTargets.Class, AllowMultiple = true)]
+    public class ContractPermissionAttribute : Attribute
+    {
+        public string HashOrGroup { get; set; }
```
Comment (erikzhang) [inline src/Neo.SmartContract.Framework/ContractPermissionAttribute.cs:8]:
```suggestion
        public string Contract { get; set; }
```

File: templates/Template.CSharp/Contract1.cs [hunk 3]
```diff
@@ -9,6 +9,7 @@ namespace $safeprojectname$
     [ManifestExtra("Author", "Neo")]
     [ManifestExtra("Email", "dev@neo.org")]
     [ManifestExtra("Description", "This is a contract example")]
+    [WildcardContractPermission("onNEP17Payment")]
```
Comment (erikzhang) [inline templates/Template.CSharp/Contract1.cs:12]: This is not a `NEP-17` contract. It doesn't need this permission.

File: src/Neo.SmartContract.Framework/WildcardContractPermissionAttribute.cs (hunk 1 above)
Comment (shargon) [reply src/Neo.SmartContract.Framework/WildcardContractPermissionAttribute.cs]: Done

File: templates/Template.CSharp/Contract1.cs (hunk 3 above)
Comment (shargon) [reply templates/Template.CSharp/Contract1.cs]: Done

**Review by erikzhang — APPROVED** [review]

**Pull request merged**